jwst/associations/lib/dms_base.py

```diff
@@ -315,8 +319,9 @@ def get_exposure_type(self, item, default='science'):
                 except KeyError:
                     pass
                 else:
-                    result = special
-                    break
+                        if exp_type not in SPEC3_SCIENCE_EXP_TYPES:
```

Comment:
This will result in any exposure in SPEC3_SCIENCE_EXP_TYPES to never be considered as background. For rules that do want such exposures to be treated as intended, this code would prevent that.